File: frontend/src/models/transaction.ts
import { Member, buildMember } from "./member";
import { MoneyUnit } from "./moneyunit";

export interface Transaction extends MoneyUnit {
  benefactor: Member;
  value: number;
}

export const buildTransaction = ({
  owner = buildMember({}),
  value = 0,
  benefactor = buildMember({})
}: Partial<Transaction>): Transaction => ({
  owner,
  value,
  benefactor
});

File: backend/src/main/java/com/github/wcosmedlr/domain/models/Transaction.java
package com.github.wcosmedlr.domain.models;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

@Data
@SuperBuilder(setterPrefix = "set", toBuilder = true)
@NoArgsConstructor
public class Transaction {

    private Member owner;
    private Double value;
    private Member benefactor;

}
